libs/community/langchain_community/callbacks/uptrain_callback.py

```diff
@@ -0,0 +1,390 @@
+"""UpTrain's Callback Handler."""
+#Langchain Community
+from collections import defaultdict
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    DefaultDict,
+    Dict,
+    List,
+    Optional,
+    Sequence,
+    Set,
+    Union,
+)
+from uuid import UUID
+
+from langchain.schema import AgentAction, AgentFinish, LLMResult
+from langchain_core.callbacks.base import AsyncCallbackHandler
+from langchain_core.documents import Document
+from langchain_core.messages.base import BaseMessage
+from uptrain import APIClient, EvalLLM, Evals, Settings
+
+
+def import_uptrain():
+    try:
+        import uptrain
+    except ImportError as e:
+        raise ImportError(
+            "To use the UpTrainCallbackHandler, you need the"
+            "`uptrain` package. Please install it with"
+            "`pip install uptrain`.",
+            e
+        )
+
+    return uptrain
+
+class UpTrainDataSchema:
+    """UpTrain Data Schema"""
+
+    def __init__(self, project_name_prefix: str) -> None:
+        """Initialize the UpTrain data schema."""
+        # For tracking project name and results
+        self.project_name_prefix: str = project_name_prefix
+        self.uptrain_results: DefaultDict[str, Any] = defaultdict(list)
+
+        # For tracking event types
+        self.eval_types: Set[str] = set()
+
+        ## RAG
+        self.query: str = ""
+        self.context: str = ""
+        self.response: str = ""
+
+        ## CONTEXT CONCISENESS
+        self.old_context: list[str] = []
+        self.new_context: list[str] = []
+        self.context_conciseness_run_id: str = ""
+
+        # MULTI QUERY
+        self.multi_queries: List[str] = []
+        self.multi_query_run_id: str = ""
+        self.multi_query_daugher_run_id: str = ""
+
+class UpTrainCallbackHandler(AsyncCallbackHandler):
+    """Callback Handler that logs to uptrain.
+
+    Args:
+        BaseCallbackHandler (_type_): _description_
+    """
+
+    REPO_URL: str = "https://github.com/uptrain-ai/uptrain"
```

Comment:
Why is this here?

Reply:
I had used https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/callbacks/argilla_callback.py for reference. I see that other callback files don't have it. I have removed it now.